yoyodyne/models/modules/lstm.py

```diff
@@ -223,3 +223,149 @@ def forward(
     @property
     def name(self) -> str:
         return "attentive LSTM"
+
+
+class HardAttentionLSTMDecoder(LSTMDecoder):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Activates emission probs.
+        self.output_proj = nn.Sequential(
+            nn.Linear(self.output_size, self.output_size), nn.Tanh()
+        )
+        # Project transition probabilities to depth of module.
+        self.scale_encoded = nn.Linear(
+            self.decoder_input_size, self.hidden_size
+        )
+
+    def _alignment_step(
```

Comment:
I think this and the overridden version of this method would benefit from a method signature.

Reply:
could you clarify? do you mean beyond typing?